proposed/2021/NukeInstallationsInstructions.md

```diff
@@ -0,0 +1,51 @@
+# Title
+
+- Author Name [Matthias Koch (@matkoch)](https://github.com/matkoch)
+- Start Date 2021-04-21
+- GitHub Issue https://github.com/NuGet/Home/issues/10784
+- GitHub PR https://github.com/NuGet/Home/pull/10782
+
+## Summary
+
+Add package installation instructions for using .NET tools with NUKE.
+
+## Motivation
+
+Conventionally, users add references to packages using the common `PackageReference` approach. This has two downsides: Firstly, it doesn't allow to reference multiple versions. Secondly, it sometimes breaks the build project, when packages ship more than just the `./tools` folder. This has been the case for `DocFX` for instance.
+
+The command helps developers to add a NuGet package in the _right_ way. So I personally expect less confusion on their side, and as a nice bonus it will also teach more folks about the existence of the `PackageDownload` feature.
+
+## Explanation
+
+### Functional explanation
+
+NUKE ships with a lot wrapper APIs for third-party tools, like `ReportGenerator`, `GitVersion.Tool` and many more. These APIs are available, but could fail at runtime, because the required package is not referenced. When I want to use the `ReportGeneratorTasks` for instance, I may head over to NuGet.org, check what particular version I need, copy the command, and execute it in my terminal from whereever I am. It doesn't have to be the build project directory. The command can be used repository-wide, as there is only one build project.
+
+### Technical explanation
+
+There are no technical details for NuGet other that we might need to set a `NUGET_USER_AGENT` or similar.
+
+On our side, the command simply adds the `PackageReference` item to the `csproj` file using `Microsoft.Build` infrastructure, and then performs a `dotnet restore` on the build project which triggers the download to the global NuGet cache. It is only converted to `PackageDownload` if the package contains a `./tools` directory.
+
+## Drawbacks
+
+NUKE is not strictly a NuGet package manager. The instructions however _are_ useful for package consumers.
+
+## Rationale and alternatives
+
+The alternative is to provide a `dotnet packagedownload` command. However, this command might be confusing in relation to the existing `dotnet package` command. Also, even compared to this hypothetical command, `nuke :add-package` covers a broader range, since it can be called from any directory in the repository, and the command will know where to add the `PackageDownload`.
+
+## Prior Art
+
+The Paket team successfully pioneered adding a new tab. The Cake team has successfully pushed [their proposal](https://github.com/NuGet/NuGetGallery/issues/8381). As I already mentioned in the issue, I'm not particularly fond of showing installation instructions for build-irrelevant packages. In any case, I'm limiting my proposal to packages with `DotnetTool` as a `packageType`. For everything else, i.e. `PackageReference`, the IDE capabilities should suffice.
```

Comment:
Just to help clarify based on the proposal here:

> In any case, I'm limiting my proposal to packages with DotnetTool as a packageType. For everything else, i.e. PackageReference, the IDE capabilities should suffice.

There was a visual proposal in the previous issue showing package dependencies. Can we assume that is cut from the scope in this proposal based on the above sentence regarding `PackageReference`? 

![image](https://user-images.githubusercontent.com/1288848/115576806-76115a00-a289-11eb-82b8-f56b5d59e738.png)

Reply:
That's correct. Initially, I thought about joining based on the responses from @joelverhagen, but I rather follow my own objections and limit the scope of this. There has been negative feedback about this already.

Reply:
@JonDouglas I need to add something. The `packageType=DotnetTool` is already available as metadata. Under `Future Possibilities` I also mentioned packages that have a `./tools` folder. I don't know how likely it is to expect metadata about that at some point in the future, but it would be great if we could consider this part of the proposal. Let me know if I should clarify that! :)